### PR TITLE
A devicelab test of memory usage for a minimal app

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@yjbanov

File: dev/devicelab/lib/tasks/perf_tests.dart [hunk 1]
```diff
@@ -185,11 +193,10 @@ class BuildTest {
 }
 
 class MemoryTest {
-  MemoryTest(this.testDirectory, this.testTarget, this.timelineFileName, this.packageName, { this.os });
+  MemoryTest(this.testDirectory, this.testTarget, this.packageName, { this.os });
```
Comment (yjbanov) [inline dev/devicelab/lib/tasks/perf_tests.dart]: Let's document what `testTarget` parameter means, and what happens when it's not specified.

File: dev/devicelab/lib/tasks/perf_tests.dart [hunk 2]
```diff
@@ -52,6 +51,15 @@ TaskFunction createComplexLayoutBuildTest() {
   return new BuildTest('${flutterDirectory.path}/dev/benchmarks/complex_layout');
 }
 
+TaskFunction createHelloWorldMemoryTest({ @required DeviceOperatingSystem os }) {
+  return new MemoryTest(
+    '${flutterDirectory.path}/examples/hello_world',
+    null,
```
Comment (yjbanov) [inline dev/devicelab/lib/tasks/perf_tests.dart]: A more idiomatic way to express an optional parameter is to use a named parameter with a default value.

**Review by yjbanov — APPROVED** [review]

File: dev/devicelab/lib/tasks/perf_tests.dart (hunk 2 above)
Comment (jason-simmons) [inline dev/devicelab/lib/tasks/perf_tests.dart]: done

File: dev/devicelab/lib/tasks/perf_tests.dart (hunk 1 above)
Comment (jason-simmons) [inline dev/devicelab/lib/tasks/perf_tests.dart]: done

File: dev/devicelab/lib/tasks/perf_tests.dart [hunk 3]
```diff
@@ -184,15 +191,18 @@ class BuildTest {
...
+  MemoryTest(this.testDirectory, this.packageName, { this.testTarget, this.os });
 
   final String testDirectory;
-  final String testTarget;
-  final String timelineFileName;
   final String packageName;
   final DeviceOperatingSystem os;
 
+  // Path to a flutter driver script that will run after starting the app.
+  // If not specified, then the test will start the app, gather statistics, and then exit.
```
Comment (yjbanov) [inline dev/devicelab/lib/tasks/perf_tests.dart]: nits: use `///` for dartdocs and make the first sentence a standalone paragraph (i.e. separate it from the next sentence by 1 blank line).

**Review by yjbanov — APPROVED** [review]

File: dev/devicelab/lib/tasks/perf_tests.dart (hunk 3 above)
Comment (jason-simmons) [inline dev/devicelab/lib/tasks/perf_tests.dart]: done

**Pull request merged**